### PR TITLE
Added EventDispatcher and event

### PR DESCRIPTION
**Pull request by mmoreram**
* When Sitemap is generated
* Created dumpers. Console command has no logic right now
* Fixed definition of Banner EventDispatcher (Now extending AbstractEventListener, the right way)
* Created chain for getting all Dumpers

### CODE REVIEW

File: src/Elcodi/Bundle/SitemapBundle/DependencyInjection/ElcodiSitemapExtension.php [hunk 1]
```diff
@@ -165,13 +178,75 @@ protected function loadCommands(array $config, ContainerBuilder $container)
...
+        $profiles = $config['profiles'];
+
+        foreach ($profiles as $profileName => $profile) {
+
+            $container
+                ->register(
+                    'elcodi.sitemap_command.' . $profileName,
+                    '%elcodi.core.sitemap.command.dump_sitemap.class%'
+                )
+                ->addArgument(new Reference('elcodi.sitemap_dumper.' . $profileName))
```
Comment (xphere) [inline src/Elcodi/Bundle/SitemapBundle/DependencyInjection/ElcodiSitemapExtension.php]: Do we really want to create one command for each sitemap?
I think it's better to have just one command and choice the sitemap with a parameter (or dump all if no parameter is set)

File: src/Elcodi/Bundle/SitemapBundle/Resources/config/classes.yml [hunk 2]
```diff
@@ -16,6 +16,17 @@ parameters:
...
+
+    #
     # Commands
     #
     elcodi.core.sitemap.command.dump_sitemap.class: Elcodi\Component\Sitemap\Command\DumpSitemapCommand
+
+    #
+    # Event Dispatchers
+    #
+    elcodi.core.sitemap.event_dispatcher.sitemap.class: Elcodi\Component\Sitemap\EventDispatcher\SitemapEventDispatcher
```
Comment (xphere) [inline src/Elcodi/Bundle/SitemapBundle/Resources/config/classes.yml]: Yaml also want their own end-of-file "\n".

File: src/Elcodi/Bundle/SitemapBundle/DependencyInjection/ElcodiSitemapExtension.php (hunk 1 above)
Comment (mmoreram) [inline src/Elcodi/Bundle/SitemapBundle/DependencyInjection/ElcodiSitemapExtension.php]: It is SNCRedis-style. Its much easier for final users to have an specific command with an specific action. If you work with parameters its more "complicated"

File: src/Elcodi/Bundle/SitemapBundle/Resources/config/classes.yml (hunk 2 above)
Comment (mmoreram) [inline src/Elcodi/Bundle/SitemapBundle/Resources/config/classes.yml]: Yep, sure, thanks

File: src/Elcodi/Component/Sitemap/Dumper/SitemapDumperChain.php [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+ */
+ 
+namespace Elcodi\Component\Sitemap\Dumper;
+
+use Elcodi\Component\Sitemap\Dumper\Interfaces\SitemapDumperInterface;
+
+/**
+ * Class SitemapDumperChain
+ */
+class SitemapDumperChain
```
Comment (xphere) [inline src/Elcodi/Component/Sitemap/Dumper/SitemapDumperChain.php]: This is also a `SitemapDumperInterface` implementing the Composite pattern.

Comment (mmoreram) [inline src/Elcodi/Component/Sitemap/Dumper/SitemapDumperChain.php]: It makes sense to `dump` all elements?

File: src/Elcodi/Bundle/SitemapBundle/DependencyInjection/ElcodiSitemapExtension.php (hunk 1 above)
Comment (mmoreram) [inline src/Elcodi/Bundle/SitemapBundle/DependencyInjection/ElcodiSitemapExtension.php]: I'll merge this version and I have created [this technical debt issue](https://github.com/elcodi/elcodi/issues/468)

Comment (xphere) [inline src/Elcodi/Bundle/SitemapBundle/DependencyInjection/ElcodiSitemapExtension.php]: :+1:

**Pull request merged**